src/dpi.erl

```diff
@@ -125,42 +108,23 @@ rpc_call(Node, Mod, Fun, Args) ->
         {badrpc, {'EXIT', Error}} ->
             error(Error);
         {badrpc, nodedown} ->
-            erase(dpi_slave),
+            erase(Node),
             error({slave_down_internal, Node, Mod, Fun, Args});
         Result ->
             Result
     end.
 
-process_res(create, State, _Args , Result) -> State ++ refList(Result);
-process_res(delete, State, Args, _Result) -> State -- refList(Args);
-process_res(_, _, _, _) -> error.
-
-refList(Input) when is_map(Input)-> refList(maps:values(Input));
-refList(Input) when is_reference(Input)-> [Input];
-refList(Input) when is_list(Input)->
-    lists:filter(
-        fun is_reference/1,
-        lists:flatten([refList(X) || X <- Input])
-    );
-refList(_Input) -> [].
-
-%% starts the gen_server and "standardizes" the return value.
-%% on success, it will always be {ok, Pid}, even if it already exsits,
-%% in which case there would be a different value that is however transformed
-%% to {ok, Pid} instead. This way the calling function doesn't have to handle
-%% this extra case every time
-gen_server_rpc_start(Slave) ->
-    Res = rpc_call(
-        Slave, gen_server, start_link,
-        [{local, ?MODULE}, ?MODULE, [], []]
-    ),
-    case Res of
-        {ok, Pid} -> {ok, Pid};
-        {error, {already_started, Pid}} -> {ok, Pid};
-        Else -> error({start_link_load, Else})
-    end.
+%%         module  fun         args       result of the call
+-spec safe(atom(), atom(),     list()) -> term().
+-spec safe(        function(), list()) -> term().
+-spec safe(        function()        ) -> term().
+
+
```

Comment:
Remove empty lines.
In general, no need for empty lines between spec and function defn!